src/CMakeLists.txt

```diff
@@ -59,6 +59,10 @@ if(NATS_BUILD_LIB_SHARED)
   set_property(TARGET nats PROPERTY DEBUG_POSTFIX d)
   target_include_directories(nats PUBLIC
         $<INSTALL_INTERFACE:include>)
+  if(NATS_BUILD_LIB_SHARED_STATIC)
+    target_link_options(nats PRIVATE
```

Comment:
Not an expert at all, so forgive the question but: why `PRIVATE` here?

Reply:
The PRIVATE here is to make this -static NOT to be inherited by the targets that decided to use nats as PUBLIC
e.g. target_link_libraries(my_3rd_party_target PUBLIC -lnats)